base_layer/wallet/src/output_manager_service/storage/sqlite_db/output_sql.rs

```diff
@@ -292,6 +299,132 @@ impl OutputSql {
+        }
+
+        query = query.then_order_by(outputs::value.asc());
+
+        let transaction_input_limit = u32::try_from(range_limit.transaction_input_limit)
+            .unwrap_or(u32::MAX)
+            .min(TRANSACTION_INPUTS_LIMIT);
+        let outputs: Vec<OutputSql> = query.limit(i64::from(transaction_input_limit)).load(conn)?;
+
+        // Post processing in Rust vs. all in raw SQL.
+        //
+        // The current version:
+        // - does a single backend query;
+        // - loads at most transaction_input_limit rows;
+        // - runs a cheap O(n) scan.
+        // The raw SQL alternative would conceptually:
+        // - order outputs and compute a running sum;
+        // - give each row a row number in that order;
+        // - find the smallest row number whose running_sum >= target_minimum_amount;
+        // - return all rows with row_number <= that row_number.
+        // The raw SQL alternative:
+        // - is more complex;
+        // - is less type-safe;
+        // - is harder to maintain/debug.
+
+        // If all the outputs together don't reach target, we cannot continue
+        let total_sum: u64 = outputs.iter().fold(0u64, |acc, o| acc.saturating_add(o.value as u64));
+        if total_sum < range_limit.target_minimum_amount {
+            debug!(
+                target: LOG_TARGET,
+                "Total unspent outputs' value in the specified range was less than the target_minimum_amount: {} < {}",
+                total_sum, range_limit.target_minimum_amount
+            );
+            return Ok((Vec::new(), MicroMinotari::zero()));
+        }
+
+        // Pick the smallest prefix that reaches the target
+        let mut selected = Vec::new();
+        let mut total = 0u64;
+        for output in outputs {
+            total = total.saturating_add(output.value as u64);
+            selected.push(output);
+            if total >= range_limit.target_minimum_amount {
+                break;
+            }
+        }
+
+        Ok((selected, MicroMinotari::from(total)))
+    }
+
+    /// Retrieves UTXO counts grouped by the provided ranges
+    pub fn count_outputs_in_ranges(
+        selection_criteria: &UtxoSelectionCriteria,
+        ranges: &[Range<u64>],
+        tip_height: Option<u64>,
+        conn: &mut SqliteConnection,
+    ) -> Result<Vec<CoinBucket>, OutputManagerStorageError> {
+        let mut result = Vec::with_capacity(ranges.len());
+        let i64_tip_height = tip_height.and_then(|h| i64::try_from(h).ok()).unwrap_or(i64::MAX);
+
+        for range in ranges {
+            let amounts_from = i64::try_from(range.start).unwrap_or(i64::MAX);
+            let amounts_to = i64::try_from(range.end).unwrap_or(i64::MAX);
+
+            let mut query = outputs::table
+                .into_boxed()
+                .filter(outputs::status.eq(OutputStatus::Unspent as i32))
+                .filter(outputs::value.ge(amounts_from))
```

Comment:
should be >= to fit the rust range defination 
By not including `=` you are exlcuding the edges of the range in counting